plugins/modules/net_tools/nmcli.py

```diff
@@ -279,6 +279,19 @@
             - When updating this property on a currently activated connection, the change takes effect immediately.
        type: str
        version_added: 2.0.0
+    wifi_sec:
+       description:
+            - 'The security configuration of the Wifi connection. The valid attributes are listed on:'
+            - 'U(https://developer.gnome.org/NetworkManager/stable/settings-802-11-wireless-security.html)'
+            - 'For instance to use common WPA-PSK auth with a password:'
+            - '- C({key-mgmt: wpa-psk, psk: my_password})'
+       type: dict
```

Comment:
It might be interesting to actually add the suboptions here, and mark the wifi passwords as `no_log`.

Reply:
If we list the `suboptions` we create a strong dependency with one specific NetworkManager version. I prefer to offer as much flexibility as possible.

Reply:
It's kind of a PITA that this is dependent on the NM version.

Maybe the module should provide one consistent interface and convert it to whatever the nmcli version in use requires?

Reply:
Why? This would complexify the logic and the module would still require a manual to support new keys.

Reply:
Mainly because it is easier for users, who can use the same task for different nm versions without having to adjust the parameters depending on the nm version. Also it allows to have a better control on the suboptions, so that only the passphrase itself is marked as `no_log`, while the other values are not. Right now, if any non-secret value in this `dict` also appears anywhere else in the module output, it will be censored. Which can potentially make the output pretty unhelpful.